ts/main.ts

```diff
@@ -262,13 +317,46 @@ function renderSearchedCards(cardObjects: any): any {
       data.cards.unshift(cardInfo);
       data.nextCardId++;
     });
+
+    // EVENT LISTENER: to listen for when a card object is clicked and checks if user wants to quick add to collection
+
+    $cardContainer.addEventListener('click', (event) => {
+      $cardDetailsAddButton.className = 'card-details-button';
+      const eventTarget = event.target as HTMLElement;
+      if (eventTarget.tagName === 'BUTTON') {
+        $addButton.addEventListener('click', () => {
+          alert('Card Added Successfully', 1000);
+          cardInfo.cardId = data.nextCardId;
+          data.cards.unshift(cardInfo);
+          data.nextCardId++;
+        });
```

Comment:
You don't want to add an event listener inside of another event listener, because now anytime you ever click on the `$cardContainer`, you will be adding multiple event listeners to the same element. So now you'd have 50 click events on the same button